Lib/multiprocessing/forkserver.py

```diff
@@ -255,8 +291,22 @@ def sigchld_handler(*_unused):
                 if listener in rfds:
                     # Incoming fork request
                     with listener.accept()[0] as s:
-                        # Receive fds from client
-                        fds = reduction.recvfds(s, MAXFDS_TO_SEND + 1)
+                        try:
+                            if authkey:
+                                wrapped_s = connection.Connection(s.fileno())
+                                try:
+                                    connection.deliver_challenge(
+                                            wrapped_s, authkey)
+                                    connection.answer_challenge(
+                                            wrapped_s, authkey)
```

Comment:
Would it be worth having a comment that note that the other half of this exchange happens in `connect_to_new_process()` (and a similar note there)?

Reply:
done.